packages/smart-accounts/src/light-account/lightAccountStaticImpl.ts

```diff
@@ -0,0 +1,193 @@
+type LightAccountV1StaticImpl = StaticSmartAccountImplementation<
+  false,
+  "0.6",
+  LightAccountFactoryArgs,
+  typeof entryPoint06Abi,
+  typeof LightAccountAbi_v1,
+  typeof LightAccountFactoryAbi_v1
+>;
+
+const lightAccountV1Base = {
+  entryPoint: {
+    abi: entryPoint06Abi,
+    address: entryPoint06Address,
+    version: "0.6",
+  },
+  accountAbi: LightAccountAbi_v1,
+  factoryAbi: LightAccountFactoryAbi_v1,
+  getFactoryData: (factoryArgs: LightAccountFactoryArgs) => {
+    return encodeFunctionData({
+      abi: LightAccountFactoryAbi_v1,
+      functionName: "createAccount",
+      args: [factoryArgs.owner, factoryArgs.salt],
+    });
+  },
+} satisfies Partial<LightAccountV1StaticImpl>;
+
+export const lightAccountStaticImplV1_0_1: LightAccountV1StaticImpl = {
+  ...lightAccountV1Base,
+  accountImplementation:
+    lightAccountVersionRegistry.LightAccount["v1.0.1"].addresses.default.impl,
+  factoryAddress:
+    lightAccountVersionRegistry.LightAccount["v1.0.1"].addresses.default
+      .factory,
+  predictAccountAddress: (factoryArgs) => {
+    return predictLightAccountAddress({
+      factoryAddress:
+        lightAccountVersionRegistry.LightAccount["v1.0.1"].addresses.default
+          .factory,
+      salt: factoryArgs.salt,
+      ownerAddress: factoryArgs.owner,
+      version: "v1.0.1",
+    });
+  },
+};
+
+export const lightAccountStaticImplV1_0_2: LightAccountV1StaticImpl = {
+  ...lightAccountV1Base,
+  accountImplementation:
+    lightAccountVersionRegistry.LightAccount["v1.0.2"].addresses.default.impl,
+  factoryAddress:
+    lightAccountVersionRegistry.LightAccount["v1.0.2"].addresses.default
+      .factory,
+  predictAccountAddress: (factoryArgs) => {
+    return predictLightAccountAddress({
+      factoryAddress:
+        lightAccountVersionRegistry.LightAccount["v1.0.2"].addresses.default
+          .factory,
+      salt: factoryArgs.salt,
+      ownerAddress: factoryArgs.owner,
+      version: "v1.0.2",
+    });
+  },
+};
+
+export const lightAccountStaticImplV1_1_0: LightAccountV1StaticImpl = {
+  ...lightAccountV1Base,
+  accountImplementation:
+    lightAccountVersionRegistry.LightAccount["v1.1.0"].addresses.default.impl,
+  factoryAddress:
+    lightAccountVersionRegistry.LightAccount["v1.1.0"].addresses.default
+      .factory,
+  predictAccountAddress: (factoryArgs) => {
+    return predictLightAccountAddress({
+      factoryAddress:
+        lightAccountVersionRegistry.LightAccount["v1.1.0"].addresses.default
+          .factory,
+      salt: factoryArgs.salt,
+      ownerAddress: factoryArgs.owner,
+      version: "v1.1.0",
+    });
+  },
+};
+
+// Shared type and fields for all light account v2 implementations
+const lightAccountV2Base = {
+  entryPoint: {
+    abi: entryPoint07Abi,
+    address: entryPoint07Address,
+    version: "0.7",
+  },
+} satisfies Partial<StaticSmartAccountImplementation<false, "0.7">>;
+
+export const lightAccountStaticImplV2_0_0: StaticSmartAccountImplementation<
```

Comment:
as a newcomer, all the different light account versions seem pretty intimidating. can we add comments or a rough guide on when to use which version?

Reply:
similarly, do we need a migration guide for devs migrating from v4 that have been creating dummy signers?